pkg/kubelet/preemption/preemption_test.go

```diff
@@ -90,82 +90,58 @@ func getTestCriticalPodAdmissionHandler(podProvider *fakePodProvider, podKiller
-			expectErr:             false,
-			expectedOutput:        nil,
-		},
-	}
-	for _, r := range runs {
-		podProvider.setPods(r.inputPods)
-		outErr := criticalPodAdmissionHandler.evictPodsToFreeRequests(allPods[clusterCritical], r.insufficientResources)
-		outputPods := podKiller.getKilledPods()
-		if !r.expectErr && outErr != nil {
-			t.Errorf("evictPodsToFreeRequests returned an unexpected error during the %s test.  Err: %v", r.testName, outErr)
-		} else if r.expectErr && outErr == nil {
-			t.Errorf("evictPodsToFreeRequests expected an error but returned a successful output=%v during the %s test.", outputPods, r.testName)
-		} else if !podListEqual(r.expectedOutput, outputPods) {
-			t.Errorf("evictPodsToFreeRequests expected %v but got %v during the %s test.", r.expectedOutput, outputPods, r.testName)
-		}
-		podKiller.clear()
-	}
-}
-
 func TestEvictPodsToFreeRequests(t *testing.T) {
 	type testRun struct {
 		testName              string
+		isPodKillerWithError  bool
 		inputPods             []*v1.Pod
 		insufficientResources admissionRequirementList
 		expectErr             bool
 		expectedOutput        []*v1.Pod
 	}
-	podProvider := newFakePodProvider()
-	podKiller := newFakePodKiller(false)
-	criticalPodAdmissionHandler := getTestCriticalPodAdmissionHandler(podProvider, podKiller)
 	allPods := getTestPods()
 	runs := []testRun{
 		{
 			testName:              "critical pods cannot be preempted",
+			isPodKillerWithError:  false,
 			inputPods:             []*v1.Pod{allPods[clusterCritical]},
 			insufficientResources: getAdmissionRequirementList(0, 0, 1),
 			expectErr:             true,
 			expectedOutput:        nil,
 		},
 		{
 			testName:              "best effort pods are not preempted when attempting to free resources",
+			isPodKillerWithError:  false,
 			inputPods:             []*v1.Pod{allPods[bestEffort]},
 			insufficientResources: getAdmissionRequirementList(0, 1, 0),
 			expectErr:             true,
 			expectedOutput:        nil,
 		},
 		{
-			testName: "multiple pods evicted",
+			testName:             "multiple pods evicted",
+			isPodKillerWithError: false,
 			inputPods: []*v1.Pod{
 				allPods[clusterCritical], allPods[bestEffort], allPods[burstable], allPods[highRequestBurstable],
 				allPods[guaranteed], allPods[highRequestGuaranteed]},
 			insufficientResources: getAdmissionRequirementList(0, 550, 0),
 			expectErr:             false,
 			expectedOutput:        []*v1.Pod{allPods[highRequestBurstable], allPods[highRequestGuaranteed]},
 		},
+		{
+			testName:             "multiple pods with eviction error",
+			isPodKillerWithError: true,
+			inputPods: []*v1.Pod{
+				allPods[clusterCritical], allPods[bestEffort], allPods[burstable], allPods[highRequestBurstable],
+				allPods[guaranteed], allPods[highRequestGuaranteed]},
+			insufficientResources: getAdmissionRequirementList(0, 550, 0),
+			expectErr:             false,
+			expectedOutput:        nil,
+		},
 	}
 	for _, r := range runs {
```

Comment:
why no `t.Run` here? Do they need to be sync?

Reply:
thanks! fixed.